core/src/main/java/io/grpc/ServiceProviders.java

```diff
@@ -71,11 +71,16 @@ private ServiceProviders() {
       list.add(current);
     }
 
-    // Sort descending based on priority.
+    // Sort descending based on priority.  If priorities are equal, compare the class names to
```

Comment:
This is contrary to "If there are more than one name-sake providers ranking the highest priority, the one registered first will be effective."

If it returns zero, then a stable sort will produce the correct order.

Reply:
> This is contrary to "If there are more than one name-sake providers ranking the highest priority, the one registered first will be effective."

I don't think so. This determines the order of providers being registered in the default register. Sorting by class name gives an deterministic registration order.